### PR TITLE
Port zero-dce example to `keras-core`

### PR DESCRIPTION
**Pull request by soumik12345**
Port the keras-io example [Zero-DCE for low-light image enhancement](https://keras.io/examples/vision/zero_dce/) to `keras_core`.

`tf.keras` backwards compatibility check: https://github.com/keras-team/keras-core/issues/485
Colab Notebook: https://colab.research.google.com/drive/1GuumdRuh9wy_Eo_zzORFfG2iX8gGG58a?usp=sharing

### CODE REVIEW

File: examples/keras_io/tensorflow/vision/zero_dce.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+
+    def train_step(self, data):
+        with tf.GradientTape() as tape:
+            output = self.dce_model(data)
+            losses = self.compute_losses(data, output)
+        gradients = tape.gradient(
+            losses["total_loss"], self.dce_model.trainable_weights
+        )
+        self.optimizer.apply_gradients(zip(gradients, self.dce_model.trainable_weights))
+        return losses
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/zero_dce.py:371]: Instead of this we should return the `result()` of a set of `Mean` trackers for each loss

File: examples/keras_io/tensorflow/vision/zero_dce.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    return output_image
+
+
+"""
+### Inference on test images
+
+We compare the test images from LOLDataset enhanced by MIRNet with images enhanced via
+the `PIL.ImageOps.autocontrast()` function.
+
+You can use the trained model hosted on [Hugging Face Hub](https://huggingface.co/keras-io/low-light-image-enhancement)
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/zero_dce.py:431]: Does that actually work with Keras Core? If not let's just drop the paragraph.

**Review by fchollet — COMMENTED** [review]
Thank you for the PR!

File: examples/keras_io/tensorflow/vision/zero_dce.py [hunk 3]
```diff
@@ -0,0 +1,442 @@
+"""
+Title: Zero-DCE for low-light image enhancement
+Author: [Soumik Rakshit](http://github.com/soumik12345)
+Converted to Keras Core by: [Soumik Rakshit](http://github.com/soumik12345)
+Date created: 2021/09/18
+Last modified: 2021/09/19
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/zero_dce.py:6]: Update date

File: examples/keras_io/tensorflow/vision/zero_dce.py (hunk 1 above)
Comment (soumik12345) [reply examples/keras_io/tensorflow/vision/zero_dce.py:371]: Hi @fchollet 
Tried [this](https://colab.research.google.com/drive/1GuumdRuh9wy_Eo_zzORFfG2iX8gGG58a?usp=sharing), however, it doesn't seem to present inside `history.history`.

File: examples/keras_io/tensorflow/vision/zero_dce.py (hunk 2 above)
Comment (soumik12345) [reply examples/keras_io/tensorflow/vision/zero_dce.py:431]: This seems to be working. Here's an inference notebook from the pre-trained model from HuggingFace hub: https://colab.research.google.com/drive/1BiLRXhpcFpkc3_J1f4YsbTA16kU4ZrJI?usp=sharing

File: examples/keras_io/tensorflow/vision/zero_dce.py (hunk 3 above)
Comment (soumik12345) [reply examples/keras_io/tensorflow/vision/zero_dce.py]: Should the `Date created` be modified as well?

**Review by fchollet — COMMENTED** [review]
Thanks! Do you confirm it's running? We're not running it on CI so you'll want to make sure it runs on your side before we merge.

**Comment by soumik12345** [comment]
> Thanks! Do you confirm it's running? We're not running it on CI so you'll want to make sure it runs on your side before we merge.

It's running perfectly, here's the [colab](https://colab.research.google.com/drive/1GuumdRuh9wy_Eo_zzORFfG2iX8gGG58a?usp=sharing) for reference.

**Review by fchollet — APPROVED** [review]
Thank you!
